overlord/devicestate/devicestate.go

```diff
@@ -1921,11 +1921,17 @@ func InstallFinish(st *state.State, label string, onVolumes map[string]*gadget.V
 	finishTask := st.NewTask("install-finish", fmt.Sprintf("Finish setup of run system for %q", label))
 	finishTask.Set("system-label", label)
 	finishTask.Set("on-volumes", onVolumes)
+	finishTask.Set("optional-install", optionalInstall)
 	chg.AddTask(finishTask)
 
 	return chg, nil
 }
 
+type OptionalInstall struct {
```

Comment:
probably worth a short doc comment, also maybe this should be defined before InstallFinish

Reply:
what about the suggestion to move it?

Reply:
Oh sorry, missed that.